src/test/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/StartBusinessProcessTaskHandlerTest.java

```diff
@@ -0,0 +1,135 @@
+    @MockBean
+    private CoreCaseDataService coreCaseDataService;
+    @Autowired
+    private StartBusinessProcessTaskHandler startBusinessProcessTaskHandler;
+
+    @BeforeEach
+    void init() {
+        when(mockExternalTask.getTopicName()).thenReturn("test");
+        when(mockExternalTask.getWorkerId()).thenReturn("worker");
+        when(mockExternalTask.getActivityId()).thenReturn("activityId");
+        when(mockExternalTask.getProcessInstanceId()).thenReturn(PROCESS_INSTANCE_ID);
+
+        when(mockExternalTask.getAllVariables())
+            .thenReturn(Map.of(
+                "caseId",
+                CASE_ID,
+                "caseEvent",
+                START_BUSINESS_PROCESS.name()
+            ));
+    }
+
+    @Test
+    void shouldUpdateBusinessStatusToStarted_whenInputStatusIsReady() {
+        CaseData caseData = new CaseDataBuilder().atStateClaimDraft()
+            .businessProcess(BusinessProcess.builder().status(BusinessProcessStatus.READY).build())
+            .build();
+
+        CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+
+        when(coreCaseDataService.startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS)))
+            .thenReturn(StartEventResponse.builder().caseDetails(caseDetails).build());
+
+        when(coreCaseDataService.submitUpdate(eq(CASE_ID), any(CaseDataContent.class)))
+            .thenReturn(caseData);
+
+        startBusinessProcessTaskHandler.execute(mockExternalTask, externalTaskService);
+
+        verify(coreCaseDataService).startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS));
+        verify(coreCaseDataService).submitUpdate(eq(CASE_ID), any(CaseDataContent.class));
+        verify(externalTaskService).complete(eq(mockExternalTask), any(VariableMap.class));
+    }
+
+    @Test
+    void shouldUpdateBusinessStatusToSarted_whenInputStatusIsDispatched() {
+        CaseData caseData = new CaseDataBuilder().atStateClaimDraft()
+            .businessProcess(BusinessProcess.builder().status(BusinessProcessStatus.DISPATCHED).build())
+            .build();
+
+        CaseDetails caseDetails = CaseDetailsBuilder.builder()
+            .data(caseData)
+            .build();
+
+        when(coreCaseDataService.startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS)))
+            .thenReturn(StartEventResponse.builder().caseDetails(caseDetails).build());
+
+        when(coreCaseDataService.submitUpdate(eq(CASE_ID), any(CaseDataContent.class)))
+            .thenReturn(caseData);
+        startBusinessProcessTaskHandler.execute(mockExternalTask, externalTaskService);
+
+        verify(coreCaseDataService).startUpdate(eq(CASE_ID), eq(START_BUSINESS_PROCESS));
+        verify(coreCaseDataService).submitUpdate(eq(CASE_ID), any(CaseDataContent.class));
+        verify(externalTaskService).complete(eq(mockExternalTask), any(VariableMap.class));
+    }
+
+    @Test
+    void shouldRaiseBpmnError_whenStatusExecutedIsStarted() {
```

Comment:
It's actually testing situation in which status is started and process instance is the same. We need separate test for STARTED + different process id to gett this covered. And one more test for asserting BpmnError for FINISHED status.